### PR TITLE
fix subcalls relation

### PR DESCRIPTION
**Pull request by tmcgroul**

### CODE REVIEW

File: substrate/substrate-processor/src/mapping.ts [hunk 1]
```diff
@@ -405,6 +405,10 @@ export function setUpItems(block: Block): void {
                 rec.parentCall = prev
                 populateSubcalls(prev, rec)
             }
+            if (prev.parentCall && isSubcall(prev.parentCall, rec)) {
```
Comment (belopash) [inline substrate/substrate-processor/src/mapping.ts:408]: nested calls are handled incorrectly.

[1, 1, 1],  [1, 1], [1]  -> the first will be child of the second, the second will be cild of the third

[1, 1, 1],  [1]  -> the first will be child of the third

Comment (tmcgroul) [reply substrate/substrate-processor/src/mapping.ts:408]: yeah, i also expect that it works like that

Comment (tmcgroul) [reply substrate/substrate-processor/src/mapping.ts:408]: but i don't see the problem...

Comment (tmcgroul) [reply substrate/substrate-processor/src/mapping.ts:408]: ok if we talk about `[1, 1, 1], [1]` then there will be wrong parent assignment...
the parent will have the child as a subcall but i believe `[1]` shouldn't be in `.parentCall`

Comment (belopash) [reply substrate/substrate-processor/src/mapping.ts:408]: idk that should be the correct behavior here, but `isSubcall` assumes that `[1, 1, 1]` is a subcall of `[1]`

Comment (tmcgroul) [reply substrate/substrate-processor/src/mapping.ts:408]: it is expected according to subcalls query - https://github.com/subsquid/archive.py/blob/master/sqa/substrate/query.py#L422-L426

File: substrate/substrate-processor/src/mapping.ts [hunk 2]
```diff
@@ -487,3 +493,8 @@ function isSubcall(parent: CallKey, call: CallKey): boolean {
     }
     return true
 }
+
+
+function traverseCalls(prev: Call, call: Call) {
```
Comment (belopash) [inline substrate/substrate-processor/src/mapping.ts:498]: ?

File: substrate/substrate-processor/src/mapping.ts (hunk 1 above)
Comment (belopash) [reply substrate/substrate-processor/src/mapping.ts]: > it is expected according to subcalls query -
Bot sure, I guess it only says "request all subcalls and thier child", but nothing about how they should be grouped

File: substrate/substrate-processor/src/mapping.ts (hunk 2 above)
Comment (tmcgroul) [reply substrate/substrate-processor/src/mapping.ts:498]: somebody did it. not me 💅

File: substrate/substrate-processor/src/mapping.ts (hunk 1 above)
Comment (tmcgroul) [reply substrate/substrate-processor/src/mapping.ts]: yeah, another way is to put only direct "children" into `subcalls` array. i guess both options are fine...
@eldargab how subcalls are supposed to work?

Comment (tmcgroul) [reply substrate/substrate-processor/src/mapping.ts]: i got confirmation that subcalls represent all children

**Pull request merged**